app/controllers/hipstergrams_controller.rb

```diff
@@ -0,0 +1,48 @@
+class HipstergramsController < ApplicationController
+  
+  before_action :authenticate_hipster!
+  before_action :check_author, :only => [:edit, :update, :destroy]
+
+  def index
+   @hipstergrams = Hipstergram.all 
+   @hipster = current_hipster
+  end
+
+  def show
+    @hipstergram = Hipstergram.find(params[:id])
+  end
+
+  def new
+    @hipstergram = Hipstergram.new
+  end
+
+  def create
+    @hipstergram = Hipstergram.new(hipstergram_params)
+    if @hipstergram.save
+      redirect_to "/hipstergrams"
+    else
+      render "new"
+    end
+  end
+
+  def hipstergram_params
+    new_params = params.require(:hipstergram).permit(:image, :description)
+    new_params[:hipster_id] = current_hipster.id
```

Comment:
Usually we would do current_user.hipstergram.new(hipstergram_params) in the controller rather than add the user_id to the params hash.